palemoon: 29.2.0 -> 29.2.1

OPNA2608 opened the pull request:
###### Motivation for this change
Just bugfixes, no CVE fixes this time.

https://repo.palemoon.org/MoonchildProductions/Pale-Moon/releases/tag/29.2.1_Release

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jonringer commented on pkgs/applications/networking/browsers/palemoon/default.nix:
```diff
@@ -202,7 +201,6 @@ stdenv.mkDerivation rec {
 
     # Remove unneeded SDK data from installation
-    # TODO: move to a separate output?
     rm -rf $out/{include,share/idl,lib/palemoon-devel-${version}}
```
should this go into a "dev" output?

OPNA2608 replied:
https://forum.palemoon.org/viewtopic.php?f=37&t=26796&p=214676#p214729

Upstream devs said it's of not much use to anyone and I should just remove it. Their prebuilt packages don't include it either.

AndersonTorres replied:
@OPNA2608 what do you think to add this info as a small comment?

OPNA2608 committed:
palemoon: 29.2.0 -> 29.2.1

OPNA2608 commented:
- Added links to upstream's forums for the SDK removal & wrong WMClass in desktop file
- Added asserts of the whole compiler requirements stuff mentioned at the top-level comment of the derivation
- Added `meta.changelog` entry pointing to the release tag on their gitea instance, which has the same content that ends up on their release notes page
